config/src/config/network_config.rs

```diff
@@ -176,12 +175,13 @@ impl NetworkConfig {
 
     pub fn random_with_peer_id(&mut self, rng: &mut StdRng, peer_id: Option<PeerId>) {
         let signing_key = Ed25519PrivateKey::generate(rng);
-        let identity_key = X25519StaticPrivateKey::generate(rng);
-        let network_keypairs = NetworkKeyPairs::load(signing_key, identity_key);
+        let identity_key = x25519::PrivateKey::for_test(rng);
```

Comment:
Can we somehow annotate or group functions that are just for testing? And obviously any other function that invokes them is also (or if it isn't it should be) for testing.

Reply:
yup that was the plan! But doing this means I have to convert a bunch of other code to test code which I wanted to do in another PR : o